packages/compass-preferences-model/src/compass-web-preferences-access.ts

```diff
@@ -0,0 +1,75 @@
+import { createNoopLogger } from '@mongodb-js/compass-logging/provider';
+import { Preferences, type PreferencesAccess } from './preferences';
+import type { UserPreferences } from './preferences-schema';
+import { type AllPreferences } from './preferences-schema';
+import { InMemoryStorage } from './preferences-in-memory-storage';
+import { getActiveUser } from './utils';
+
+export class CompassWebPreferencesAccess implements PreferencesAccess {
+  private _preferences: Preferences;
+  constructor(preferencesOverrides?: Partial<AllPreferences>) {
+    this._preferences = new Preferences({
+      logger: createNoopLogger(),
+      preferencesStorage: new InMemoryStorage(preferencesOverrides),
+    });
+  }
+
+  savePreferences(_attributes: Partial<UserPreferences>) {
+    if (
+      Object.keys(_attributes).length === 1 &&
+      'optInDataExplorerGenAIFeatures' in _attributes
+    ) {
+      return Promise.resolve(this._preferences.savePreferences(_attributes));
+    }
+    // do not save any attributes other than the optInDataExplorerGenAIFeatures setting
```

Comment:
Small nit, avoiding negation to make the comment a bit quicker to read:
```suggestion
    // Only allow saving the optInDataExplorerGenAIFeatures preference.
    if (
      Object.keys(_attributes).length === 1 &&
      'optInDataExplorerGenAIFeatures' in _attributes
    ) {
      return Promise.resolve(this._preferences.savePreferences(_attributes));
    }
```